srcCpp/RTIDDSImpl.cxx

```diff
@@ -594,6 +594,30 @@ class RTIPublisherBase : public IMessagingWriter
       #endif
     }
 
+    unsigned int getSampleCount()
+    {
+      #ifndef RTI_MICRO
+        DDS_DataWriterCacheStatus status;
+        _writer->get_datawriter_cache_status(status);
```

Comment:
Check for errors and propagate them. We should throw an exception but we do not handle them properly for the moment.
I know that this would make the code difficult but I don't see any other option.

Reply:
I thought about this... but I don't really know what to propagate... as you said it should be an exception. For the time being I will just display a message.

Reply:
Done.